src/PowerShellEditorServices/Session/Host/EditorServicesPSHostUserInterface.cs

```diff
@@ -642,8 +642,8 @@ public override void WriteErrorLine(string value)
                     .OfType<string>()
                     .FirstOrDefault() ?? "PS> ";
 
-            // Add the [DBG] prefix if we're stopped in the debugger
-            if (this.powerShellContext.IsDebuggerStopped)
+            // Add the [DBG] prefix if we're stopped in the debugger and the prompt doesn't already have [DBG] in it
+            if (this.powerShellContext.IsDebuggerStopped && !promptString.Contains("[DBG]"))
```

Comment:
Do we need the check at all or could we just remove the "add DBG to string" logic entirely?

Reply:
I think we could improve the current logic but a PowerShell host typically does add `[DBG]: ` to the beginning of the prompt during debug.  However, PowerShell.exe will not do this if it has detected the user has a custom prompt function.  For now, I think this is reasonable - avoid duplicate `[DBG]` in the prompt string.